opal/resource.go

```diff
@@ -428,6 +439,62 @@ func resourceResourceRead(ctx context.Context, d *schema.ResourceData, m any) di
 			remoteInfoIList = append(remoteInfoIList, map[string]any{
 				"aws_permission_set": permissionSetIList,
 			})
+		case opal.RESOURCETYPEENUM_AWS_IAM_ROLE:
+			var metadata map[string]opal.ResourceRemoteInfoAwsIamRole
```

Comment:
figured out the bug - so for IAM roles and EKS clusters the metadata doesn't match the remote info format...

unsure if the best way forward is to make the metadata field names match or to make a new API object, doing this for now @ken-opal

Reply:
i kinda wanna just put a todo here to make it consistent

Reply:
were you able to validate that the workflow of running terraformer first to import some IAM roles, and then editing the TF for the role resource and running `tf apply` is successful with this change? curious because like you mentioned the metadata for the role has account name instead of ID, which is in the remote info:

![image](https://github.com/opalsecurity/terraform-provider-opal/assets/106631529/f7beee7e-e6c6-42ef-9935-3f25bbcc9b3c)

if this workflow works fine, it sgtm. agreed that we probably shouldn't change the stored metadata structure

Reply:
otherwise, I think I would prefer to just have terraformer not populate the `remote_info` for these types

Reply:
another alternative - manually define a structs here to match the JSON metadata responses, and then for the missing `account_id`s we can pass in `resource.ParentResourceId` which is available from the response obj